src/vaadin-dialog.html

```diff
@@ -289,6 +391,172 @@
             e.preventDefault();
           }
         }
+
+        _addDraggableEventListeners() {
+          const overlay = this.$.overlay.$.overlay;
+          const originalBounds = {top: 0, left: 0};
+          const originalMouse = {top: 0, left: 0};
+          const stopDrag = () => {
+            window.removeEventListener('mouseup', stopDrag);
+            window.removeEventListener('mousemove', drag);
+          };
+          const drag = (e) => {
+            this._snapToPixels();
+            const top = originalBounds.top + (e.pageY - originalMouse.top);
+            const left = originalBounds.left + (e.pageX - originalMouse.left);
+            this._setBounds({top, left});
+          };
+          const startDrag = (e) => {
+            if (this.draggable) {
+              const target = e.target;
+              const isOverlayPart = target.matches ? target.matches('[part="overlay"]') : target.msMatchesSelector('[part="overlay"]');
```

Comment:
This looks fragile because:
- There might be content in the overlay that actually has `part="content"` (maybe from some other vaadin component), this would cause unintended behavior
- It falls back to `target.msMatchesSelector` if `target.matches` isn't found
- The selector `[part="overlay"]` requires that the element doesn't have other part names.

Rather use (still not ideal but maybe less fragile):
```js
const isOverlayPart = e.target === this.$.overlay.$.overlay;
const isContentPart = e.target === this.$.overlay.$.content;
```

Reply:
Done.